emission/analysis/modelling/trip_model/support_vector_machine.py

```diff
@@ -0,0 +1,117 @@
+import emission.analysis.modelling.trip_model.util as eamtu
+
+
+class SupportVectorMachine(eamuu.TripModel):
+
+    is_incremental: bool = False  # overwritten during __init__
+    incremental_classes: list = None # overwritten when fit is initialized
+
+    def __init__(self, config=None):
+        """
+        Instantiate a linear support vector machine for all users.
+
+        This uses the sklearn implementation of a support vector machine 
+        to classify unlabeled replacement modes. The SVM is linear, and is fit 
+        with the more general SGDClassifier class which can accommodate online 
+        learning:
+
+        https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html
+
+        For anyone looking to implement a differnt online learning model in the 
+        future, here is a list of sklearn models that implement "partial_fit" 
+        and would be candidates for the online learning approach implemented 
+        here:
+
+        https://scikit-learn.org/0.15/modules/scaling_strategies.html
+
+        Replacement modes are considered to be the second-best choice for
+        a given trip (i.e., what mode would have been chosen if the actual
+        choice wasn't available).
+
+        The model is currently trained on data from all users.
+        """
+        if config is None:
+            config = eamtc.get_config_value_or_raise('model_parameters.svm')
+            logging.debug(f'SupportVectorMachine loaded model config from file')
+        else:
+            logging.debug(f'SupportVectorMachine using model config argument')
+        expected_keys = [
+            'incremental_evaluation',
+            'feature_list',
+            'dependent_var'
+        ]
+        for k in expected_keys:
+            if config.get(k) is None:
+                msg = f"svm trip model config missing expected key {k}"
+                raise KeyError(msg)
+        self.is_incremental = config['incremental_evaluation']
+        # use the sklearn implementation of a svm
+        self.svm = SGDClassifier()
+        self.feature_list = config['feature_list']
+        self.dependent_var = config['dependent_var']
+
+    def fit(self, trips: List[ecwc.Confirmedtrip]):
+        """train the model by passing data, where each row in the data
+        corresponds to a label at the matching index of the label input.
+
+        If using an incremental model, the initial call to fit will store 
+        the list of unique classes in y. If additional classes are added in the 
+        future, the model will need to be re-initialized on data that 
+        contains all of those classes, otherwise sklearn will throw an 
+        error. Additionally, any categorical variables used as features 
+        must have all classes present in the initial training data. If not, 
+        they will throw an sklearn error due to differences in the number of 
+        OHE variables in the data.
```

Comment:
Initial training set: [walk, bike]
new data comes in with: [drive], so we retrain, and new training set has [walk, bike, drive]
new data comes in with: [walk], but because the current model has [walk, bike, drive], it again throws an error